File: bab_11_class/Sarjana.py
#Simpan dengan nama file Sarjana.py
from Student import Student

class Sarjana(Student):
    def __init__(self):
        self.jurusan = ""
        self.gelar = ""

    def add_data_lulusan(self, new_id, new_name, new_grade, new_jurusan, new_gelar):
        self.add_data(new_id, new_name, new_grade)
        self.jurusan = new_jurusan
        self.gelar = new_gelar

    # Method untuk menampilkan detail mahasiswa sarjana
    def display(self):
        print("Undergraduate Student")
        print("ID:", self.id, ", Name:", self.nama, ", Grade:", self.grade)
        print("Jurusan:", self.jurusan)
        print("Gelar:", self.gelar)

File: bab_11_class/Sarjana.cpp
//Simpan dengan nama file Sarjana.cpp
#include <iostream>
#include <string>
#include "Sarjana.h"

// Implementasi method untuk mengisi data sarjana
void Sarjana::addDataLulusan(int newId, const std::string& newName, 
    char newGrade, const std::string& newJurusan, 
    const std::string& newGelar) {
    addData(newId, newName, newGrade);
    jurusan = newJurusan;
    gelar = newGelar;
}

// Implementasi method untuk menampilkan detail mahasiswa sarjana
void Sarjana::display() {
    std::cout << "Undergraduate Student" << std::endl;
    std::cout << "ID: " << id << ", Name: " << nama << ", Grade: " << grade << std::endl;
    std::cout << "Jurusan: " << jurusan << std::endl;
    std::cout << "Gelar: " << gelar << std::endl;
}
